token-group/js/src/instruction.ts

```diff
@@ -42,10 +37,10 @@ export function createInitializeGroupInstruction(args: InitializeGroupInstructio
             { isSigner: true, isWritable: false, pubkey: mintAuthority },
         ],
         data: packInstruction(
-            [
+            getStructEncoder([
                 ['updateAuthority', getBytesEncoder({ size: 32 })],
                 ['maxSize', getU32Encoder()],
-            ],
+            ]),
             splDiscriminate('spl_token_group_interface:initialize_token_group'),
             { updateAuthority: updateAuthorityBuffer, maxSize }
```

Comment:
Since you're using the old web3.js here and cannot use the `getAddressEncoder` helper from `@solana/addresses`, I'd recommend creating your own `getPublicKeyEncoder` helper:

```ts
function getPublicKeyEncoder(): Encoder<PublicKey> {
    return mapEncoder(
        getBytesEncoder({ size: 32 }),
        (publicKey: PublicKey) => publicKey.toBytes()
    );
}

// Usage.
getPublicKeyEncoder().encode(updateAuthority);

// Usage if updateAuthority must be replaces by 32 zero bytes if missing.
getPublicKeyEncoder().encode(updateAuthority ?? SystemProgram.programId);
```

Reply:
Gotcha, will do!